tests/console/redis.pm

```diff
@@ -20,65 +20,166 @@ use serial_terminal 'select_serial_terminal';
+    my (%args) = @_;
+    $args{port} //= $default_redis_port;
+
+    test_ping(port => $args{port});
+    test_crud(port => $args{port});
+    load_test_db_and_validate(port => $args{port});
+}
+
+sub configure_and_test_replica {
+    my (%args) = @_;
+    $args{port} //= $default_redis_port;
+    $args{replica_port} //= $default_redis_replica_port;
+
+    my $redis_replica_cli_cmd = "redis-cli -p $args{replica_port}";
+
+    test_ping(port => $args{replica_port});
+
+    # Configure replication
+    assert_script_run("$redis_replica_cli_cmd replicaof localhost $args{port}");
+
+    verify_replication_status();
+
+    # Validate data from the replica
+    validate_script_output("$redis_replica_cli_cmd HMGET \"movie:343\" title", sub { m/Spider-Man/ });
+}
 
-    validate_script_output('redis-cli HMGET "movie:343" title', sub { m/Spider-Man/ });
+sub cleanup_redis {
+    my (%args) = @_;
+    $args{redis_version} //= $default_redis_version;
+    $args{port} //= $default_redis_port;
+    $args{replica_port} //= $default_redis_replica_port;
 
-    # start redis server on port 6380 and test that it works
-    assert_script_run('redis-server --daemonize yes --port 6380 --logfile /var/log/redis/redis-server_6380.log');
-    validate_script_output_retry('redis-cli -p 6380 ping', sub { m/PONG/ }, delay => 5, retry => 12);
+    my $redis_cli_cmd_prefix = "redis-cli -p ";
+    my $redis_cli_cmd_postfix = " flushall";
 
-    # make 6380 instance a replica of redis instance running on port 6379
-    assert_script_run('redis-cli -p 6380 replicaof localhost 6379');
+    my $redis_cli_cmd = $redis_cli_cmd_prefix . $args{port} . $redis_cli_cmd_postfix;
+    my $redis_replica_cli_cmd = $redis_cli_cmd_prefix . $args{replica_port} . $redis_cli_cmd_postfix;
 
-    # test master knows about the slave and vice versa
-    validate_script_output_retry('redis-cli info replication', sub { m/connected_slaves:1/ }, delay => 5, retry => 12);
-    validate_script_output('redis-cli -p 6380 info replication', sub { m/role:slave/ });
+    # Clean up after testing
+    assert_script_run($redis_cli_cmd);
+    assert_script_run($redis_replica_cli_cmd);
 
-    # test that the synchronization finished and the data are reachable from slave
-    validate_script_output_retry('redis-cli info replication', sub { m/state=online/ }, delay => 5, retry => 12);
-    validate_script_output('redis-cli -p 6380 HMGET "movie:343" title', sub { m/Spider-Man/ });
+    assert_script_run($killall_redis_server_cmd);
+    assert_script_run($remove_test_db_file_cmd);
+    zypper_call('rm -u ' . $args{redis_version});
+}
+
+sub test_redis {
+    my (%args) = @_;
+    $args{redis_version} //= $default_redis_version;
+    $args{port} //= $default_redis_port;
+    $args{replica_port} //= $default_redis_replica_port;
+
+    zypper_call('in --force-resolution --solver-focus Update ' . $args{redis_version});
+
+    my ($version) = script_output('redis-server --version') =~ (/v=(.+?)\s/)[0] || "";
```

Comment:
If version is "" test should die, the variable is not crucial, but it's still better to be valid, whatever changes we will know.